pyxem/signals/symmetry1d.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2020 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Signal class for pearson correlation and symmetry coefficient."""
+
+from hyperspy.signals import Signal2D, BaseSignal, Signal1D
+#from hyperspy._signals.lazy import LazySignal
+import numpy as np
+
+
+class Symmetry1D(Signal1D):
+    _signal_type = "symmetry"
+
+    def __init__(self, *args, **kwargs):
+        """Create a Symmetry object from a numpy.ndarray.
+
+        Parameters
+        ----------
+        *args :
+            Passed to the __init__ of Signal1D. The first arg should be
+            a numpy.ndarray
+        **kwargs :
+            Passed to the __init__ of Signal1D
+        """
+        super().__init__(*args, **kwargs)
+
+        self.decomposition.__func__.__doc__ = BaseSignal.decomposition.__doc__
+
+    def get_symmetry_coefficient(self):
+        """Return symmetry coefficient from pearson correlation function at all real
+        space positions (n from 2 to 10).
+
+        Returns
+        -------
+        sn: Signal1D
+            Symmetry coefficient
+        """
+        sn = Signal1D(np.zeros(self.axes_manager._navigation_shape_in_array+(11,)))
+        sn.isig[2] = self.isig[np.pi]
```

Comment:
Pretty sure `.isig[x] `needs `x` to be an integer? I'm not sure how this passes the tests. Update: (see review comment)

Reply:
I think the slicing method in hyperspy allows `.isig[x]` to take in float argument. If `x` is a float, it would treat it as value of real physical unit instead of array index.

Reply:
see discussion, considered resolved.